lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1916,22 +1916,36 @@ void TypeChecker::checkIgnoredExpr(Expr *E) {
       .highlight(E->getSourceRange());
     return;
   }
-    
-  // Always complain about 'try?'.
-  if (auto *OTE = dyn_cast<OptionalTryExpr>(valueE)) {
-    DE.diagnose(OTE->getTryLoc(), diag::expression_unused_optional_try)
-      .highlight(E->getSourceRange());
-    return;
-  }
 
   if (auto *LE = dyn_cast<LiteralExpr>(valueE)) {
     diagnoseIgnoredLiteral(Context, LE);
     return;
   }
+  
+  ApplyExpr *call = nullptr;
+  
+  // Unwrap optional try to get inner function calls.
```

Comment:
The function call expr of `try? foo()` is wrapped in `OptionalTryExpr` and `InjectIntoOptionalExpr`. It need to unwrap them.

```
(optional_try_expr type="Int?" location=issue.swift:5:8 range=[issue.swift:5:3 - line:5:12] thrown_error="<null>"
  (inject_into_optional implicit type="Int?" location=issue.swift:5:8 range=[issue.swift:5:8 - line:5:12]
    (call_expr type="Int" location=issue.swift:5:8 range=[issue.swift:5:8 - line:5:12] isolation_crossing="none"
      (declref_expr type="() throws -> Int" location=issue.swift:5:8 range=[issue.swift:5:8 - line:5:8] decl="issue.(file).foo()@issue.swift:2:6" function_ref=single)
      (argument_list))))
```